File: Coding Platforms/Hacker Earth/practice/barclays/MaxMul.java
import java.io.*;
import java.util.*;


 class MaxMul {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter wr = new PrintWriter(System.out);
        int T = Integer.parseInt(br.readLine().trim());

        for(int t_i=0; t_i<T; t_i++)
        {   int N = Integer.parseInt(br.readLine().trim());
            String[] arr_A = br.readLine().split(" ");
            
            int[] A = new int[N];
            for(int i_A=0; i_A<arr_A.length; i_A++)
            {
            	A[i_A] = Integer.parseInt(arr_A[i_A]);
            }
            
            long out_ = solve(A,N);
            System.out.println(out_);
            
         }

         wr.close();
         br.close();
    }
    static long solve(int[] A,int N){
        //Write your code here
        long prod1 = 0;
        long prod2 = 0;

        if (N == 1) 
		    return (long)A[0]; 

        if (N == 2) {
            return (Math.abs(A[0]) * Math.abs(A[1])); 
        } 

        Arrays.sort(A);

        prod1 = Math.abs(A[N-2]) * Math.abs(A[N-1]);
        prod2 = Math.abs(A[0]) * Math.abs(A[1]);

        if(prod1 > prod2){
            return prod1;
        }else{
            return prod2;
        }
         
    }
}

File: Coding Platforms/Hacker Earth/practice/barclays/MaxMul.cpp
#include<bits/stdc++.h>
using namespace std;

long long solve(vector<int> a,int n) {
   //Write your code here
   /*  int i;
    int largest1 = a[0];
   
    for (i = 0; i < n; i++) {
        if (a[i] > largest1) {
            largest1 = a[i];
            cout << largest1 << " L1 ";
        }
    }
    
    int largest2 = a[0];
    
    for (i = 1; i < n; i++) {
        if (a[i] > largest2 && a[i] < largest1)
            largest2 = a[i];
            cout << largest2 << " L2 ";
    } */

    for (int i = 0; i < n; i++)
    {
        for (int  j = 0; j < n; j++)
        {
            
            

        }
        
    }
    
   
}

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int T;
    cin >> T;
    
    for(int t_i=0; t_i<T; t_i++)
    {   int N;
        cin >> N;
        vector<int> A(N);
        for(int i_A=0; i_A<N; i_A++)
        {
        	cin >> A[i_A];
        }
        long long out_;
        out_ = solve(A,N);
        cout << out_;
        cout << "\n";
    }
}